### PR TITLE
Upgrade to Django 4.2

### PR DESCRIPTION
**Pull request by Viicos**

### CODE REVIEW

**Comment by Viicos** [comment]
Next steps would be:
- Remove `drf-yasg`: it seems to be compatible with 4.2 (although only 4.1 is explicitly stated as supported), but is badly maintained and will not work on Python 3.12
- Figure out what to do with `iso-639`: will also not work with 3.12, not maintained anymore. Alternative: https://github.com/jacksonllee/iso639

File: vng_api_common/inspectors/query.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 try:
     from django.utils.encoding import force_str
 except ImportError:  # Django < 4.0
-    from django.utils.encoding import force_text as force_str
+    from django.utils.encoding import force_str as force_str
```
Comment (annashamray) [inline vng_api_common/inspectors/query.py:6]: I think it should all be reduced to 1 line:
`from django.utils.encoding import force_str`

File: vng_api_common/caching/etags.py [hunk 2]
```diff
@@ -157,7 +157,8 @@ def mark_affected(cls, obj: models.Model, using=None) -> None:
         connection = transaction.get_connection(using)
 
         func = MethodCallback(etag_update.calculate_new_value)
-        for _, _func in connection.run_on_commit:
+        for run_on_commit in connection.run_on_commit:
+            _func = run_on_commit[1]
```
Comment (annashamray) [inline vng_api_common/caching/etags.py:161]: Why is this changed needed?

**Review by annashamray — CHANGES_REQUESTED** [review]
ON + django4.2 - checked, works without errors
OZ + django3.2 - checked, there are some errors with validator
OZ + django4.2 - not checked, because it requires quite some work to update OZ itself and drc-cmis, before it can be tested. But I don't think it should block this PR. 

So after minor remarks are addressed, it's good to go

File: vng_api_common/validators.py [hunk 3]
```diff
@@ -311,40 +304,33 @@ class UniekeIdentificatieValidator:
...
-        pk = self.instance.pk if self.instance else None
+        pk = instance.pk if instance else None
 
         # if we're updating an instance, setting the current values will not
         # trigger an error because the instance-to-be-updated is excluded from
         # the queryset. If either bronorganisatie or identificatie changes,
         # and it already exists, it will raise a validation error
         combination_exists = (
-            self.model.objects
+            serializer.Meta.model
```
Comment (annashamray) [inline vng_api_common/validators.py:333]: This validator doesn't work anymore and fails with this error (for Besluit model):
`AttributeError: type object 'Besluit' has no attribute 'exclude'` 

I think the reason is this line
```suggestion
            serializer.Meta.model.objects

File: vng_api_common/caching/etags.py (hunk 2 above)
Comment (Viicos) [reply vng_api_common/caching/etags.py:161]: `run_on_commit` is private Django API, and change to be a two-tuple to a three-tuple. The previous `for x, y in ...` did not work anymore

Comment (annashamray) [reply vng_api_common/caching/etags.py:161]: Thanks for the explanation!

**Review by annashamray — APPROVED** [review]
